neet/boolean/randomnet.py

```diff
@@ -25,6 +25,48 @@ def mean_degree(net):
 from neet.sensitivity import canalizing_nodes
 from .logicnetwork import LogicNetwork
 
+def RBN(network_size, k, p, self_loops=False):
+    """
+    Given a network with $n$ nodes, the dynamic state of each node $i$
+    at time $t+1$ depends on $k$ randomly selected input nodes
+    $i_1,\dots,i_k$ at time $t$.
+    
+    There are $2^k$ possible configurations of
+    $[ x_{i_1}(t),\dots,x_{i_k}(t) ]$, and each one of them
+    coresponds to $x_i(t+1)=1$ with probability $p$.
+    """
+    connectivity,bias = k,p
+    new_table = []
+    for output_label in range(network_size):
+        one_states = []
+        inputs = _RBN_input_labels(output_label, network_size, connectivity,
+                                   self_loops=self_loops)
+        inputs.sort()
+        pattern = _RBN_output_pattern(connectivity,bias)
+        for j in range(len(pattern)):
+            if pattern[j]==1:
+                one_states.append(list(_label_to_state(j,connectivity)))
+        new_table.append([inputs,one_states])
+    return LogicNetwork(new_table)
+
+def _RBN_input_labels(output_label, network_size, connectivity, self_loops=False):
+    labels = range(network_size)
+    if not self_loops:
+        labels.remove(output_label)
+    random.shuffle(labels)
+    return labels[:connectivity]
+
+def _RBN_output_pattern(connectivity,bias):
+    pattern = []
+    for i in range(pow(2,connectivity)):
+        if random.uniform(0,1) < bias:
+            pattern.append(1)
+        else:
+            pattern.append(0)
+    return pattern
```

Comment:
Could this be done with a binomial distribution?
```
return numpy.random.binomial(n=1, p=bias, size=2**connectivity)
```

Reply:
On second thought, I don't think this will work. We'd need something like a `numpy.random.bernoulli`.